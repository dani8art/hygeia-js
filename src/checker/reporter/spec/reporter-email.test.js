'use strict';
const { EmailReporter } = require('../reporter-email');

describe('Reporters - Email', () => {

    test('Create email reporter', () => {
        let reporter = new EmailReporter({
            email: 'test@test.com'
        });

        expect(reporter.email).toBe('test@test.com');
    });

    test('Throw error when any email is passed', () => {
        expect(() => {
            let reporter = new EmailReporter({});
        }).toThrow('Email in EmailReporter options is require.');
    });

    test('Test method buildSESOptions', () => {
        let reporter = new EmailReporter({
            email: 'test@test.com'
        });
        let statusChecking = [{ service: "statuschecker", startTs: "2017-12-20T18:48:37.981Z", endTs: "2017-12-20T18:48:38.918Z", result: 200, duration: 937 }];

        let sesOptions = reporter.buildSESOptions(statusChecking);

        expect(sesOptions.Destination.ToAddresses).toContain('test@test.com');
        expect(sesOptions.Message.Body.Html.Charset).toBe('UTF-8');
        expect(sesOptions.Message.Body.Html.Data).toMatchSnapshot();
        expect(sesOptions.Message.Subject.Charset).toBe('UTF-8');
        expect(sesOptions.Message.Subject.Data).toBe('Report [lambda-health-checker]');
        expect(sesOptions.Source).toBe('checker@darteaga.com');
    });
});