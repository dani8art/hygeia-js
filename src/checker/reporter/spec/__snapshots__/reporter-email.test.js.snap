// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reporters - Email Test method buildSESOptions 1`] = `
"
            <!DOCTYPE html>
            <html>
                <head>
                
            <link rel=\\"stylesheet\\" href=\\"https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/css/bootstrap.min.css\\" integrity=\\"sha384-PsH8R72JQ3SOdhVi3uxftmaW6Vc51MKb0q5P2rRUpPvrszuE4W1povHYgTpBfshb\\" crossorigin=\\"anonymous\\">
            <style>
                .card-title{text-transform: uppercase;}
                .card{margin-bottom: 5px;}
                h6{font-size: 80%;font-weight: 400;}
            </style>    
        
                </head>
                <body>
                
            <div class=\\"container\\">
                <h3>Health report</h3>
                <div class=\\"row\\">
                    
                <div class=\\"col-12\\">
                    <div class=\\"card\\">
                        <div class=\\"card-body\\">
                            <h5 class=\\"card-title\\">statuschecker <span class=\\"float-right badge badge-pill badge-success\\">200</span></h5>
                            <h6 class=\\"card-subtitle mb-2 text-muted\\">2017-12-20 18:48:37</h6>
                            <p class=\\"card-text\\">
                                <b>Duration: </b> 937 ms
                            </p>
                        </div>
                    </div>
                </div>
            
                </div>
            </div>
        
                </body>
            </html>
        "
`;
